### PR TITLE
V3.0.7/actualize api

### PR DESCRIPTION
**Pull request by alexions**
Checkout API: Actualize models.

Customer Token API: Actualize models. 
 - TokenErrorV2 model: new field authorized_payment_method_v1
 - TokenErrorV2 model: new field fraud_status
 - TokenErrorV2 model: new field reason

HPP API: Actualize models.
HPPMerchantUrlsV1 model:
- update examples
- add field 'back'
- add field 'error'
- add field 'status_update'

HPPResponseEntity model (breaking changes):
- Change 'StatusCodeEnum' model. Add details to the enum names:
    _100 -> _100_CONTINUE
    _101 -> _101_SWITCHING_PROTOCOLS
    _102 -> _102_PROCESSING
    ...
    _200 -> _200_OK
    _201 -> _201_CREATED
    _204 -> _204_NO_CONTENT
    ...
    _301 -> _301_MOVED_PERMANENTLY
    ...
    _400 -> _400_BAD_REQUEST
    _403 -> _403_FORBIDDEN
    ...
    _503 -> _503_SERVICE_UNAVAILABLE
    ...

CardService API: Actualize models.
New models:
- CardServiceCardSpecification
- CardServicePromiseCreatedResponse
- CardServicePromiseRequest
- CardServicePromiseResponse

Order Management API: Actualize models.

Payments API: Actualize models.
PaymentsErrorV2 model:
- add field "authorized_payment_method_v1" (PaymentsAuthorizedPaymentMethod model)
- add field "fraud_status"
- add field "reason"

Settlements API: Actualize models.

Virtual Credit Card API: 
- Mark VirtualCreditCardApi as depricated
- Add support of Virtual Credit Card Promises (with examples)
- Move VirtualCreditCardApi to VirtualCreditCardSettlementsApi

Add support of BODY payload when using DELETE method.

Instant Shopping API:  
 - Add support of declining order.
 - Update the model to the latest version.
 - Add Button Keys examples.
 - Add Unit tests

### CODE REVIEW

File: src/main/java/com/klarna/rest/Client.java [hunk 1]
```diff
@@ -108,14 +112,38 @@ public HPPSessionsApi newHPPSessionsApi() {
     /**
      * Creates a new VirtualCreditCardApi resource instance.
      *
-     * @see examples.MerchantCardServiceExample
+     * @deprecated As of 3.0.7, renaming in Merchant Card Service API.
```
Comment (freethan) [inline src/main/java/com/klarna/rest/Client.java]: Consider using 3.1.0 here as the amount of changes warrant it

File: src/test/java/com/klarna/rest/api/instant_shopping/InstantShoppingOrdersApiTest.java [hunk 2]
```diff
@@ -0,0 +1,192 @@
+/*
+ * Copyright 2018 Klarna AB
```
Comment (freethan) [inline src/test/java/com/klarna/rest/api/instant_shopping/InstantShoppingOrdersApiTest.java]: 2019

File: src/main/java/examples/InstantShoppingExample.java [hunk 3]
```diff
@@ -185,4 +183,152 @@ public static void main(final String[] args) {
...
+                                .type(InstantShoppingOrderLineV1.TypeEnum.PHYSICAL)
+                                .reference("19-402-USA")
+                                .quantity(5L)
+                                .quantityUnit("pcs")
+                                .taxRate(1000L)
+                                .totalAmount(50000L)
+                                .totalDiscountAmount(0L)
+                                .totalTaxAmount(5000L)
+                                .unitPrice(10000L)
+                                .productUrl("https://www.estore.com/products/f2a8d7e34")
```
Comment (freethan) [inline src/main/java/examples/InstantShoppingExample.java]: estore.com -> example.com

File: src/main/java/com/klarna/rest/http_transport/HttpUrlConnectionTransport.java [hunk 4]
```diff
@@ -189,13 +189,17 @@ public ApiResponse patch(final String path, final byte[] data, Map<String, Strin
     /**
      * Sends HTTP DELETE request to specified path.
      *
+     * @deprecated As of 3.0.7, adding `data` params to delete in order to cover the InstantShoppingApi
```
Comment (freethan) [inline src/main/java/com/klarna/rest/http_transport/HttpUrlConnectionTransport.java]: 3.0.7 -> 3.1.0

File: src/test/java/com/klarna/rest/api/instant_shopping/InstantShoppingOrdersApiTest.java [hunk 5]
```diff
@@ -150,9 +150,28 @@ public void testRetrieveAuthorizedOrder() throws IOException {
...
+        when(transport.conn.getHeaderFields()).thenReturn(new HashMap<String, List<String>>() {{
+            put("Content-Type", Arrays.asList(MediaType.APPLICATION_JSON));
+        }});
+
+        Client client = new Client(transport);
+        InstantShoppingOrdersApi api = client.newInstantShoppingOrdersApi("auth-token-123");
+
+        InstantShoppingMerchantDeclineOrderRequestV1 reason = new InstantShoppingMerchantDeclineOrderRequestV1()
+                .denyCode("out_of_stock")
+                .denyMessage("The item you try to buy is out of stock")
```
Comment (freethan) [inline src/test/java/com/klarna/rest/api/instant_shopping/InstantShoppingOrdersApiTest.java]: Please change to "The item you are trying to buy is out of stock"

File: src/main/java/com/klarna/rest/http_transport/HttpTransport.java [hunk 6]
```diff
@@ -123,12 +123,28 @@
     /**
      * Sends HTTP DELETE request to specified path.
      *
+     *  @deprecated As of 3.0.7, adding `data` params to delete in order to cover the InstantShoppingApi
```
Comment (freethan) [inline src/main/java/com/klarna/rest/http_transport/HttpTransport.java]: Please change all occurrences of 3.0.7 to 3.1.0

File: src/main/java/com/klarna/rest/api/instant_shopping/model/InstantShoppingAddressV1.java [hunk 7]
```diff
@@ -1,6 +1,6 @@
...
...n Europe or in the U.S.  All interaction with Klarna's APIs must be done over HTTPS. To authenticate with Klarna you use your API Credentials and HTTP Basic auth.  The credentials consist of two elements:  Username - Consists of your Merchant ID (eid) - a unique number that identifies your e-store, combined with a random string.  Password - a string which is associated with your Merchant ID and is used to authorize use of Klarna's APIs  Note: The API credentials are separate from your Merchant ID, and it is possible to have several API credentials associated with the same Merchant ID.  The credentials should be sent as an authorization header for every request with the username and password.  You receive both your live and test credentials by contacting Klarna sales for your country.  If the credentials are missing or wrong Klarna will respond with 401 Unauthorized. You can read more about HTTP Basic auth in the Wikipedia article. The username and password the username:password in base64. [https://gist.github.com/brandonmwest/a2632d0a65088a20c00a](Examples)   You can read more about Klarna Data types at https://developers.klarna.com/api/#data-types, e.g. the date format is ISO 8601
```
Comment (freethan) [inline src/main/java/com/klarna/rest/api/instant_shopping/model/InstantShoppingAddressV1.java]: > Mostly spacing and wording. I suppose this is auto-generated so this text lacks formatting. Changes have to be made in the original source file.

 * The Instant Shopping API is serving two purposes:  to manage orders as they result from the Instant Shopping purchase flow and to generate Instant Shopping Button keys necessary for setting up the Instant Shopping flow both onsite and offsite. The 'authorizations' endpoint is relevant to authorizing orders. The 'buttons' endpoint is relevant to managing button setup options. Note that as soon as a purchase initiated through Instant Shopping is authorized with Klarna, the order should be read and handled using the Order Management API. Use the Button Setup Options library for creating and maintaining Instant Shopping Button Keys. The API is accessible through a variety of URLs. There are different URLs for testing and for making live purchases as well as different URLs depending on if you are based in Europe or in the U.S. All interaction with Klarna's APIs must be done over HTTPS. To authenticate with Klarna you have to use your API Credentials and HTTP Basic auth. Credentials is comprised of two elements: Username - Consists of your Merchant ID (eid) - a unique number that identifies your e-store, combined with a random string. Password - a string which is associated with your Merchant ID and is used to authorize use of Klarna's APIs  Note: The API credentials are separate from your Merchant ID, and it is possible to have several API credentials associated with the same Merchant ID. The credentials should be sent as an authorization header for every request with the username and password. You receive both your live and test credentials by contacting Klarna sales for your country. If the credentials are missing or are wrong, Klarna will respond with 401 Unauthorized. You can read more about HTTP Basic auth in the Wikipedia article. The username and password are passed as a username:password pair, encoded in base64. [https://gist.github.com/brandonmwest/a2632d0a65088a20c00a](Examples). You can read more about Klarna Data types at https://developers.klarna.com/api/#data-types

File: src/main/java/com/klarna/rest/api/instant_shopping/model/InstantShoppingAttachmentV1.java [hunk 8]
```diff
@@ -1,6 +1,6 @@
...
...n Europe or in the U.S.  All interaction with Klarna's APIs must be done over HTTPS. To authenticate with Klarna you use your API Credentials and HTTP Basic auth.  The credentials consist of two elements:  Username - Consists of your Merchant ID (eid) - a unique number that identifies your e-store, combined with a random string.  Password - a string which is associated with your Merchant ID and is used to authorize use of Klarna's APIs  Note: The API credentials are separate from your Merchant ID, and it is possible to have several API credentials associated with the same Merchant ID.  The credentials should be sent as an authorization header for every request with the username and password.  You receive both your live and test credentials by contacting Klarna sales for your country.  If the credentials are missing or wrong Klarna will respond with 401 Unauthorized. You can read more about HTTP Basic auth in the Wikipedia article. The username and password the username:password in base64. [https://gist.github.com/brandonmwest/a2632d0a65088a20c00a](Examples)   You can read more about Klarna Data types at https://developers.klarna.com/api/#data-types, e.g. the date format is ISO 8601
```
Comment (freethan) [inline src/main/java/com/klarna/rest/api/instant_shopping/model/InstantShoppingAttachmentV1.java]: See the comment above

File: src/main/java/com/klarna/rest/api/instant_shopping/model/InstantShoppingButtonSetupOptionsV1.java [hunk 9]
```diff
@@ -1,6 +1,6 @@
...
...n Europe or in the U.S.  All interaction with Klarna's APIs must be done over HTTPS. To authenticate with Klarna you use your API Credentials and HTTP Basic auth.  The credentials consist of two elements:  Username - Consists of your Merchant ID (eid) - a unique number that identifies your e-store, combined with a random string.  Password - a string which is associated with your Merchant ID and is used to authorize use of Klarna's APIs  Note: The API credentials are separate from your Merchant ID, and it is possible to have several API credentials associated with the same Merchant ID.  The credentials should be sent as an authorization header for every request with the username and password.  You receive both your live and test credentials by contacting Klarna sales for your country.  If the credentials are missing or wrong Klarna will respond with 401 Unauthorized. You can read more about HTTP Basic auth in the Wikipedia article. The username and password the username:password in base64. [https://gist.github.com/brandonmwest/a2632d0a65088a20c00a](Examples)   You can read more about Klarna Data types at https://developers.klarna.com/api/#data-types, e.g. the date format is ISO 8601
```
Comment (freethan) [inline src/main/java/com/klarna/rest/api/instant_shopping/model/InstantShoppingButtonSetupOptionsV1.java]: See the comment above

File: src/main/java/com/klarna/rest/api/instant_shopping/model/InstantShoppingButtonSetupOptionsV1MerchantUrls.java [hunk 10]
```diff
@@ -1,6 +1,6 @@
...
...n Europe or in the U.S.  All interaction with Klarna's APIs must be done over HTTPS. To authenticate with Klarna you use your API Credentials and HTTP Basic auth.  The credentials consist of two elements:  Username - Consists of your Merchant ID (eid) - a unique number that identifies your e-store, combined with a random string.  Password - a string which is associated with your Merchant ID and is used to authorize use of Klarna's APIs  Note: The API credentials are separate from your Merchant ID, and it is possible to have several API credentials associated with the same Merchant ID.  The credentials should be sent as an authorization header for every request with the username and password.  You receive both your live and test credentials by contacting Klarna sales for your country.  If the credentials are missing or wrong Klarna will respond with 401 Unauthorized. You can read more about HTTP Basic auth in the Wikipedia article. The username and password the username:password in base64. [https://gist.github.com/brandonmwest/a2632d0a65088a20c00a](Examples)   You can read more about Klarna Data types at https://developers.klarna.com/api/#data-types, e.g. the date format is ISO 8601
```
Comment (freethan) [inline src/main/java/com/klarna/rest/api/instant_shopping/model/InstantShoppingButtonSetupOptionsV1MerchantUrls.java]: See the comment above

File: src/main/java/com/klarna/rest/api/instant_shopping/model/InstantShoppingButtonSetupOptionsV1Styling.java [hunk 11]
```diff
@@ -0,0 +1,92 @@
...
...n Europe or in the U.S.  All interaction with Klarna's APIs must be done over HTTPS. To authenticate with Klarna you use your API Credentials and HTTP Basic auth.  The credentials consist of two elements:  Username - Consists of your Merchant ID (eid) - a unique number that identifies your e-store, combined with a random string.  Password - a string which is associated with your Merchant ID and is used to authorize use of Klarna's APIs  Note: The API credentials are separate from your Merchant ID, and it is possible to have several API credentials associated with the same Merchant ID.  The credentials should be sent as an authorization header for every request with the username and password.  You receive both your live and test credentials by contacting Klarna sales for your country.  If the credentials are missing or wrong Klarna will respond with 401 Unauthorized. You can read more about HTTP Basic auth in the Wikipedia article. The username and password the username:password in base64. [https://gist.github.com/brandonmwest/a2632d0a65088a20c00a](Examples)   You can read more about Klarna Data types at https://developers.klarna.com/api/#data-types, e.g. the date format is ISO 8601
```
Comment (freethan) [inline src/main/java/com/klarna/rest/api/instant_shopping/model/InstantShoppingButtonSetupOptionsV1Styling.java]: See the comment above

File: src/main/java/com/klarna/rest/api/checkout/model/CheckoutOptions.java [hunk 12]
```diff
@@ -363,6 +345,24 @@ public void setAdditionalMerchantTerms(String additionalMerchantTerms) {
     this.additionalMerchantTerms = additionalMerchantTerms;
   }
 
+  public CheckoutOptions phoneMandatory(Boolean phoneMandatory) {
+    this.phoneMandatory = phoneMandatory;
+    return this;
+  }
+
+   /**
+   * If true, the consumer can not skip phone (in countries were this usually is possible). Only available for orders for DACH countries.
```
Comment (freethan) [inline src/main/java/com/klarna/rest/api/checkout/model/CheckoutOptions.java]: Applies also to the @ApiModelProperty line below, probably needs to be propagated to the original data source.

"If set to true, the consumer cannot avoid providing their phone number (even in countries where this is usually possible). Only available for orders made for DACH countries (Germany, Austria, Switzerland)"

File: src/main/java/com/klarna/rest/api/customer_token/model/TokenAddress.java [hunk 13]
```diff
@@ -1,9 +1,9 @@
 /*
- * Customer Tokens
- * The Customer Token API is used to charge customers with a tokenized Klarna payment method and can be used for recurring purchases, subscriptions and for storing a customer's payment method. Tokens are created using the [generate a customer token](#payments-api-generate-a-consumer-token) call in the payments API.
+ *
```
Comment (freethan) [inline src/main/java/com/klarna/rest/api/customer_token/model/TokenAddress.java]: Regression comparing to previous version ?
Please compare comments

File: src/main/java/com/klarna/rest/api/customer_token/model/TokenAttachment.java [hunk 14]
```diff
@@ -1,9 +1,9 @@
 /*
- * Customer Tokens
- * The Customer Token API is used to charge customers with a tokenized Klarna payment method and can be used for recurring purchases, subscriptions and for storing a customer's payment method. Tokens are created using the [generate a customer token](#payments-api-generate-a-consumer-token) call in the payments API.
+ * 
+ * No description provided (generated by Swagger Codegen https://github.com/swagger-api/swagger-codegen)
```
Comment (freethan) [inline src/main/java/com/klarna/rest/api/customer_token/model/TokenAttachment.java]: See above comment on regression

File: src/main/java/com/klarna/rest/api/customer_token/model/TokenAuthorizedPaymentMethod.java [hunk 15]
```diff
@@ -1,9 +1,9 @@
 /*
- * Customer Tokens
- * The Customer Token API is used to charge customers with a tokenized Klarna payment method and can be used for recurring purchases, subscriptions and for storing a customer's payment method. Tokens are created using the [generate a customer token](#payments-api-generate-a-consumer-token) call in the payments API.
+ * 
+ * No description provided (generated by Swagger Codegen https://github.com/swagger-api/swagger-codegen)
```
Comment (freethan) [inline src/main/java/com/klarna/rest/api/customer_token/model/TokenAuthorizedPaymentMethod.java]: See above comment on regression

File: src/main/java/com/klarna/rest/api/customer_token/model/TokenCardInformation.java [hunk 16]
```diff
@@ -1,9 +1,9 @@
 /*
- * Customer Tokens
- * The Customer Token API is used to charge customers with a tokenized Klarna payment method and can be used for recurring purchases, subscriptions and for storing a customer's payment method. Tokens are created using the [generate a customer token](#payments-api-generate-a-consumer-token) call in the payments API.
+ * 
+ * No description provided (generated by Swagger Codegen https://github.com/swagger-api/swagger-codegen)
```
Comment (freethan) [inline src/main/java/com/klarna/rest/api/customer_token/model/TokenCardInformation.java]: Regression in comment compared to previous version

File: src/main/java/com/klarna/rest/api/customer_token/model/TokenCustomerTokenOrder.java [hunk 17]
```diff
@@ -1,9 +1,9 @@
 /*
- * Customer Tokens
- * The Customer Token API is used to charge customers with a tokenized Klarna payment method and can be used for recurring purchases, subscriptions and for storing a customer's payment method. Tokens are created using the [generate a customer token](#payments-api-generate-a-consumer-token) call in the payments API.
+ * 
+ * No description provided (generated by Swagger Codegen https://github.com/swagger-api/swagger-codegen)
```
Comment (freethan) [inline src/main/java/com/klarna/rest/api/customer_token/model/TokenCustomerTokenOrder.java]: Regression in comment compared to previous version

File: src/main/java/com/klarna/rest/api/customer_token/model/TokenCustomerTokenStatusUpdateRequest.java [hunk 18]
```diff
@@ -1,9 +1,9 @@
 /*
- * Customer Tokens
- * The Customer Token API is used to charge customers with a tokenized Klarna payment method and can be used for recurring purchases, subscriptions and for storing a customer's payment method. Tokens are created using the [generate a customer token](#payments-api-generate-a-consumer-token) call in the payments API.
+ * 
+ * No description provided (generated by Swagger Codegen https://github.com/swagger-api/swagger-codegen)
```
Comment (freethan) [inline src/main/java/com/klarna/rest/api/customer_token/model/TokenCustomerTokenStatusUpdateRequest.java]: Regression in comment compared to previous version

File: src/main/java/com/klarna/rest/api/customer_token/model/TokenCustomerTokenV1.java [hunk 19]
```diff
@@ -1,9 +1,9 @@
 /*
- * Customer Tokens
- * The Customer Token API is used to charge customers with a tokenized Klarna payment method and can be used for recurring purchases, subscriptions and for storing a customer's payment method. Tokens are created using the [generate a customer token](#payments-api-generate-a-consumer-token) call in the payments API.
+ * 
+ * No description provided (generated by Swagger Codegen https://github.com/swagger-api/swagger-codegen)
```
Comment (freethan) [inline src/main/java/com/klarna/rest/api/customer_token/model/TokenCustomerTokenV1.java]: Regression in comment compared to previous version

File: src/main/java/com/klarna/rest/api/customer_token/model/TokenDirectDebitInformation.java [hunk 20]
```diff
@@ -1,9 +1,9 @@
 /*
- * Customer Tokens
- * The Customer Token API is used to charge customers with a tokenized Klarna payment method and can be used for recurring purchases, subscriptions and for storing a customer's payment method. Tokens are created using the [generate a customer token](#payments-api-generate-a-consumer-token) call in the payments API.
+ * 
+ * No description provided (generated by Swagger Codegen https://github.com/swagger-api/swagger-codegen)
```
Comment (freethan) [inline src/main/java/com/klarna/rest/api/customer_token/model/TokenDirectDebitInformation.java]: Regression in comment compared to previous version

File: src/main/java/com/klarna/rest/api/customer_token/model/TokenErrorV2.java [hunk 21]
```diff
@@ -1,9 +1,9 @@
 /*
- * Customer Tokens
- * The Customer Token API is used to charge customers with a tokenized Klarna payment method and can be used for recurring purchases, subscriptions and for storing a customer's payment method. Tokens are created using the [generate a customer token](#payments-api-generate-a-consumer-token) call in the payments API.
+ * 
+ * No description provided (generated by Swagger Codegen https://github.com/swagger-api/swagger-codegen)
```
Comment (freethan) [inline src/main/java/com/klarna/rest/api/customer_token/model/TokenErrorV2.java]: Regression in comment compared to previous version

File: src/main/java/com/klarna/rest/api/customer_token/model/TokenMerchantUrls.java [hunk 22]
```diff
@@ -1,9 +1,9 @@
 /*
- * Customer Tokens
- * The Customer Token API is used to charge customers with a tokenized Klarna payment method and can be used for recurring purchases, subscriptions and for storing a customer's payment method. Tokens are created using the [generate a customer token](#payments-api-generate-a-consumer-token) call in the payments API.
+ * 
+ * No description provided (generated by Swagger Codegen https://github.com/swagger-api/swagger-codegen)
```
Comment (freethan) [inline src/main/java/com/klarna/rest/api/customer_token/model/TokenMerchantUrls.java]: Regression in comment compared to previous version

File: src/main/java/com/klarna/rest/api/customer_token/model/TokenOrder.java [hunk 23]
```diff
@@ -1,9 +1,9 @@
 /*
- * Customer Tokens
- * The Customer Token API is used to charge customers with a tokenized Klarna payment method and can be used for recurring purchases, subscriptions and for storing a customer's payment method. Tokens are created using the [generate a customer token](#payments-api-generate-a-consumer-token) call in the payments API.
+ * 
+ * No description provided (generated by Swagger Codegen https://github.com/swagger-api/swagger-codegen)
```
Comment (freethan) [inline src/main/java/com/klarna/rest/api/customer_token/model/TokenOrder.java]: Regression in comment compared to previous version

File: src/main/java/com/klarna/rest/api/customer_token/model/TokenOrderLine.java [hunk 24]
```diff
@@ -1,9 +1,9 @@
 /*
- * Customer Tokens
- * The Customer Token API is used to charge customers with a tokenized Klarna payment method and can be used for recurring purchases, subscriptions and for storing a customer's payment method. Tokens are created using the [generate a customer token](#payments-api-generate-a-consumer-token) call in the payments API.
+ * 
+ * No description provided (generated by Swagger Codegen https://github.com/swagger-api/swagger-codegen)
```
Comment (freethan) [inline src/main/java/com/klarna/rest/api/customer_token/model/TokenOrderLine.java]: Regression in comment compared to previous version

File: src/main/java/com/klarna/rest/api/customer_token/model/TokenProductIdentifiers.java [hunk 25]
```diff
@@ -1,9 +1,9 @@
 /*
- * Customer Tokens
- * The Customer Token API is used to charge customers with a tokenized Klarna payment method and can be used for recurring purchases, subscriptions and for storing a customer's payment method. Tokens are created using the [generate a customer token](#payments-api-generate-a-consumer-token) call in the payments API.
+ * 
+ * No description provided (generated by Swagger Codegen https://github.com/swagger-api/swagger-codegen)
```
Comment (freethan) [inline src/main/java/com/klarna/rest/api/customer_token/model/TokenProductIdentifiers.java]: Regression in comment compared to previous version

File: src/main/java/com/klarna/rest/api/hosted_payment_page/model/HPPMerchantUrlsV1.java [hunk 26]
```diff
@@ -24,33 +24,60 @@
...
   @JsonProperty("terms")
   private String terms = null;
 
+  public HPPMerchantUrlsV1 back(String back) {
+    this.back = back;
+    return this;
+  }
+
+   /**
+   * Back URL
```
Comment (freethan) [inline src/main/java/com/klarna/rest/api/hosted_payment_page/model/HPPMerchantUrlsV1.java]: Can the authors provide a more descriptive comment here ?

**Comment by freethan** [comment]
I propose that we bump the version to 3.1.0, given the amount and nature of changes

**Comment by alexions** [comment]
> I propose that we bump the version to 3.1.0, given the amount and nature of changes

Yes, I will do a 3.1.0 since we have a breaking changes.

File: src/main/java/com/klarna/rest/Client.java (hunk 1 above)
Comment (alexions) [reply src/main/java/com/klarna/rest/Client.java]: done

File: src/main/java/com/klarna/rest/http_transport/HttpUrlConnectionTransport.java (hunk 4 above)
Comment (alexions) [reply src/main/java/com/klarna/rest/http_transport/HttpUrlConnectionTransport.java]: done

File: src/main/java/com/klarna/rest/http_transport/HttpTransport.java (hunk 6 above)
Comment (alexions) [reply src/main/java/com/klarna/rest/http_transport/HttpTransport.java]: done

File: src/test/java/com/klarna/rest/api/instant_shopping/InstantShoppingOrdersApiTest.java (hunk 2 above)
Comment (alexions) [reply src/test/java/com/klarna/rest/api/instant_shopping/InstantShoppingOrdersApiTest.java]: done

File: src/main/java/examples/InstantShoppingExample.java (hunk 3 above)
Comment (alexions) [reply src/main/java/examples/InstantShoppingExample.java]: done

File: src/test/java/com/klarna/rest/api/instant_shopping/InstantShoppingOrdersApiTest.java (hunk 5 above)
Comment (alexions) [reply src/test/java/com/klarna/rest/api/instant_shopping/InstantShoppingOrdersApiTest.java]: done

File: src/main/java/com/klarna/rest/api/instant_shopping/model/InstantShoppingAddressV1.java (hunk 7 above)
Comment (alexions) [reply src/main/java/com/klarna/rest/api/instant_shopping/model/InstantShoppingAddressV1.java]: All the models are auto-generated. But in this case: the problem with Swagger codegen and markdown generator. The generator has limited amount of markdown features. In this case the original file looks like this:

```
The Instant Shopping API is serving two purposes:\n\n to manage the orders as they result ...
```

So swagger just trimmed the new line symbols.

> Changes have to be made in the original source file

Not applicable.

File: src/main/java/com/klarna/rest/api/instant_shopping/model/InstantShoppingButtonSetupOptionsV1.java (hunk 9 above)
Comment (alexions) [reply src/main/java/com/klarna/rest/api/instant_shopping/model/InstantShoppingButtonSetupOptionsV1.java]: non-fixable

File: src/main/java/com/klarna/rest/api/instant_shopping/model/InstantShoppingButtonSetupOptionsV1Styling.java (hunk 11 above)
Comment (alexions) [reply src/main/java/com/klarna/rest/api/instant_shopping/model/InstantShoppingButtonSetupOptionsV1Styling.java]: non-fixable

File: src/main/java/com/klarna/rest/api/instant_shopping/model/InstantShoppingAttachmentV1.java (hunk 8 above)
Comment (alexions) [reply src/main/java/com/klarna/rest/api/instant_shopping/model/InstantShoppingAttachmentV1.java]: non-fixable

File: src/main/java/com/klarna/rest/api/instant_shopping/model/InstantShoppingButtonSetupOptionsV1MerchantUrls.java (hunk 10 above)
Comment (alexions) [reply src/main/java/com/klarna/rest/api/instant_shopping/model/InstantShoppingButtonSetupOptionsV1MerchantUrls.java]: non-fixable

File: src/main/java/com/klarna/rest/api/checkout/model/CheckoutOptions.java (hunk 12 above)
Comment (alexions) [reply src/main/java/com/klarna/rest/api/checkout/model/CheckoutOptions.java]: All the models are auto-generated. This change should be propagated to the original team.

File: src/main/java/com/klarna/rest/api/customer_token/model/TokenAddress.java (hunk 13 above)
Comment (alexions) [reply src/main/java/com/klarna/rest/api/customer_token/model/TokenAddress.java]: All models are auto-generated. By the reason the swagger source lost the description and other info/meta fields. There is an issue to fix this, but for now - this issue is unfixable.

File: src/main/java/com/klarna/rest/api/customer_token/model/TokenAttachment.java (hunk 14 above)
Comment (alexions) [reply src/main/java/com/klarna/rest/api/customer_token/model/TokenAttachment.java]: non-fixable

File: src/main/java/com/klarna/rest/api/customer_token/model/TokenCardInformation.java (hunk 16 above)
Comment (alexions) [reply src/main/java/com/klarna/rest/api/customer_token/model/TokenCardInformation.java]: non-fixable

File: src/main/java/com/klarna/rest/api/customer_token/model/TokenMerchantUrls.java (hunk 22 above)
Comment (alexions) [reply src/main/java/com/klarna/rest/api/customer_token/model/TokenMerchantUrls.java]: non-fixable

File: src/main/java/com/klarna/rest/api/customer_token/model/TokenOrder.java (hunk 23 above)
Comment (alexions) [reply src/main/java/com/klarna/rest/api/customer_token/model/TokenOrder.java]: non-fixable

File: src/main/java/com/klarna/rest/api/customer_token/model/TokenProductIdentifiers.java (hunk 25 above)
Comment (alexions) [reply src/main/java/com/klarna/rest/api/customer_token/model/TokenProductIdentifiers.java]: non-fixable

File: src/main/java/com/klarna/rest/api/hosted_payment_page/model/HPPMerchantUrlsV1.java (hunk 26 above)
Comment (alexions) [reply src/main/java/com/klarna/rest/api/hosted_payment_page/model/HPPMerchantUrlsV1.java]: They should. But in the case of SDK - all models are auto-generated and depend on the original source. The issue cannot be solved by SDK itself.

File: src/main/java/com/klarna/rest/api/customer_token/model/TokenOrderLine.java (hunk 24 above)
Comment (alexions) [reply src/main/java/com/klarna/rest/api/customer_token/model/TokenOrderLine.java]: non-fixable

File: src/main/java/com/klarna/rest/api/customer_token/model/TokenAuthorizedPaymentMethod.java (hunk 15 above)
Comment (alexions) [reply src/main/java/com/klarna/rest/api/customer_token/model/TokenAuthorizedPaymentMethod.java]: non-fixable

File: src/main/java/com/klarna/rest/api/customer_token/model/TokenCustomerTokenOrder.java (hunk 17 above)
Comment (alexions) [reply src/main/java/com/klarna/rest/api/customer_token/model/TokenCustomerTokenOrder.java]: non-fixable

File: src/main/java/com/klarna/rest/api/customer_token/model/TokenCustomerTokenV1.java (hunk 19 above)
Comment (alexions) [reply src/main/java/com/klarna/rest/api/customer_token/model/TokenCustomerTokenV1.java]: non-fixable

File: src/main/java/com/klarna/rest/api/customer_token/model/TokenDirectDebitInformation.java (hunk 20 above)
Comment (alexions) [reply src/main/java/com/klarna/rest/api/customer_token/model/TokenDirectDebitInformation.java]: non-fixable

File: src/main/java/com/klarna/rest/api/customer_token/model/TokenErrorV2.java (hunk 21 above)
Comment (alexions) [reply src/main/java/com/klarna/rest/api/customer_token/model/TokenErrorV2.java]: non-fixable

File: src/main/java/com/klarna/rest/api/customer_token/model/TokenCustomerTokenStatusUpdateRequest.java (hunk 18 above)
Comment (alexions) [reply src/main/java/com/klarna/rest/api/customer_token/model/TokenCustomerTokenStatusUpdateRequest.java]: non-fixable

**Pull request merged**